src/core/SuccessResponse.ts

```diff
@@ -11,4 +11,21 @@ export type Meta<T> = {
     requestHeaders: { contentType: string | null }
 }
 
-export type SuccessResponse<T> = T & { _meta: Meta<T> }
+type AsObject<T> = T extends object ? (T extends any[] ? { asObject: T } : T) : { asObject: T }
+export type SuccessResponse<T> = AsObject<T> & { _meta: Meta<T> }
+
+function isNotPureObject(value: any): boolean {
```

Comment:
nitpick: I'd think `unknown` would be better than `any`, many errors can go unnoticed with `any` and they're not there with `unknown`